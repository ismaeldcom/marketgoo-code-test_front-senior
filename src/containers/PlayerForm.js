import React, { useState } from 'react'
import { useDispatch } from 'react-redux'
import { unwrapResult } from '@reduxjs/toolkit'
import { useForm } from 'react-form'
import { Button } from '@marketgoo/ola'
import InputField from 'components/InputField'
import { addPlayer } from 'store/players'
import validation from 'helpers/validation'
import styles from './PlayerForm.module.css'

const PlayerForm = () => {
    const dispatch = useDispatch()

    const defaultValues = React.useMemo(
        () => ({
            name: '',
            team: '',
            score: ''
        }),
        []
    )
    const {
        Form,
        meta: { canSubmit, isSubmitting }
    } = useForm({
        defaultValues,
        onSubmit: (values, instance) =>
            dispatch(addPlayer(values))
                .then(unwrapResult)
                .then(() => instance.reset())
    })

    return (
        <Form className={styles.form}>
            <InputField
                field='name'
                label='Name'
                placeholder='Player name'
                validate={validation.required}
            />
            <InputField
                field='team'
                label='Team'
                placeholder='Team name'
                validate={validation.required}
            />
            <InputField
                field='score'
                label='Score'
                type='number'
                placeholder='Team Score'
                validate={validation.required}
                min='0'
            />
            <Button
                type='submit'
                variant='primary'
                className={styles.button}
                disabled={!canSubmit}
                busy={isSubmitting ? 'Add' : false}
            >
                Add
            </Button>
        </Form>
    )
}

export default PlayerForm
