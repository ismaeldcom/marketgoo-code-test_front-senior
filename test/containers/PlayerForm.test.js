import React from 'react'
import renderer from 'react-test-renderer'
import PlayerForm from 'containers/PlayerForm'

jest.mock('react-redux', () => ({
    useDispatch: () => jest.fn(),
    useSelector: jest.fn()
}))

describe('Snapshots', () => {
    describe('PlayerForm', () => {
        it('renders', () => {
            const tree = renderer.create(<PlayerForm />).toJSON()
            expect(tree).toMatchSnapshot()
        })
    })
})
