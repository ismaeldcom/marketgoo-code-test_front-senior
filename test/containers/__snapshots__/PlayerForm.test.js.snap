// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshots PlayerForm renders 1`] = `
<form
  className="form"
  disabled={false}
  onSubmit={[Function]}
>
  <div
    className="ola_field"
  >
    <label
      className="ola_field-label"
      htmlFor="field-name"
    >
      Name
    </label>
    <div
      className="ola_field-input"
    >
      <input
        className="ola_input"
        disabled={false}
        id="field-name"
        label="Name"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="Player name"
        type="text"
        value=""
      />
    </div>
  </div>
  <div
    className="ola_field"
  >
    <label
      className="ola_field-label"
      htmlFor="field-team"
    >
      Team
    </label>
    <div
      className="ola_field-input"
    >
      <input
        className="ola_input"
        disabled={false}
        id="field-team"
        label="Team"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="Team name"
        type="text"
        value=""
      />
    </div>
  </div>
  <div
    className="ola_field"
  >
    <label
      className="ola_field-label"
      htmlFor="field-score"
    >
      Score
    </label>
    <div
      className="ola_field-input"
    >
      <input
        className="ola_input"
        disabled={false}
        id="field-score"
        label="Score"
        min="0"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="Team Score"
        type="number"
        value=""
      />
    </div>
  </div>
  <button
    className="ola_button is-primary button"
    disabled={false}
    type="submit"
  >
    <span
      className="ola_button-text"
    >
      Add
    </span>
  </button>
</form>
`;
